edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/TwinHandler.cs

```diff
@@ -190,36 +204,36 @@ async Task<bool> HandleUpstreamRequest(Func<IDeviceListener, string, Task> actio
             }
 
             var message = new EdgeMessage.Builder(publishInfo.Payload).Build();
-            await action(listener, rid.Value);
+            _ = action(listener, rid.Value);
```

Comment:
this gets call when a twin requested or a reported property update gets sent. The previous before the change, it was waiting for the 'result' (e.g. when the twin manager handled the request), however now we are in the message pump, and this waiting blocks the message pump.
these operations can finish with two outcomes:
- the operation succeeds, in this case the message pump can proceed
- the operation fails. in this case the message pump swallows the exception (it reports something), but other than that it ignores the error, because it is not the message pumps concern to act on errors (the message pump just gets the mqtt packets and forwards them to handlers)

because of this, the message pump has no reason to wait for the result of the operation, it just blocks other operations to execute. In case of twin operations (get twin/reported update, which is affected by this change), there is a built in error handling in the messaging scheme, because they have a request-id, and iothub (or twin manager) needs to send an answer with that id. If the not-awaited operation fails, then these response will not be sent, so the client sooner or later will note it (that no response arrived)

On the other hand, stopping the message pump caused a problem with twin operations, when we used upstream through the broker:
- the device sent a get-twin
- the 'await' above blocked the message pump
- the twin request was forwarded to iothub, which responded with the twin
- this twin was send through the broker and the broker forwarded the message to edgeHub-bridge
- the edgeHub bridge uses the same message pump to receive this twin that is now blocked by the await.
- this way the 'await' will never be completed, because the response twin cannot come (it will be completed after timeout)

because of that, the operation could not work. Removing the 'await' unblocks the pump, so the response twin will be handled. When it arrives, edgeHub sends the response message to the client, so the 'ok' operations finish that way. If there is an error, then the missing response will lead to retries - or the necessary error handling at client side.